src/utils/I18nManager.ts

```diff
@@ -26,6 +27,16 @@ export default class I18nManager {
     }
   }
 
+  // Get the already existing instance of I18nManager depending on the locale if it exists otherwise we create it
+  public static getInstanceForLocale(locale: string): I18nManager {
+    if (!I18nManager.instances.has(locale)) {
+      I18nManager.instances.set(locale, new I18nManager(locale));
+    }
+
+    return I18nManager.instances.get(locale);
+  }
+
+
```

Comment:
@ClaudeROSSI @jerome-benoit @LucasBrazi06 here is how it ends up (now we call I18nManager. getInstanceForLocale(locale) and it either chose the existing instance or it creates it)